spec/lib/services/pollers-api-service-spec.js

```
@@ -0,0 +1,250 @@
+
+            waterline.workitems.find.resolves(mockPoller);
+            return pollerService.getPollers().then(function (pollers) {
+                expect(pollers).to.deep.equal(mockPoller);
+            });
+        });
+        it('should return error if poller informations is not found', function () {
+            var mockPollerError = new Errors.NotFoundError("Could not find workitem with identifier");
+            waterline.workitems.find.rejects(mockPollerError);
+            return pollerService.getPollers().should.eventually.be.rejectedWith(mockPollerError);
+        });
+
+    });
+
+    describe("getPollersById", function() {
+        it("should expose the appropriate methods", function() {
+            pollerService.should.have.property("getPollersById")
+            .that.is.a("function").with.length(1);
+        });
+
+        it("Run getPollersById", function() {
+            var mockPoller = [{
+                id: "1234",
+                name: "Pollers.IPMI",
+                config: {}
+            },
+            {
+                id: "4532",
+                name: "Pollers.TEST",
+                config: {}
+            }];
+            waterline.workitems.needByIdentifier.withArgs({"id": "4532"}).resolves(mockPoller[0]);
+            waterline.workitems.needByIdentifier.withArgs({"id": "1234"}).resolves(mockPoller[1]);
+            Promise.all(["4532", "1234"].map(function(id, index) {
+                return pollerService.getPollersById(index)})).then(function (pollers) {
+                    expect(pollers).to.deep.equal(mockPoller[index]);
+                });
+         });
+
+        it("should return error if specific poller info is not found", function () {
+            var mockPollerError = new Errors.NotFoundError("Could not find workitem with identifier");
+            waterline.workitems.needByIdentifier.rejects(mockPollerError);
+            return pollerService.getPollersById().should.eventually.be.rejectedWith(mockPollerError);
```

Ignoring this error.